### ISSUE TITLE
Ternary operator bool?someVoid():default gives "Failed to emit module".

### ISSUE THREAD

**Issue by nl-x** [dotnet/roslyn#41741]
**Version Used**: 
DotNetCore, any version. Not sure about other frameworks.

**Steps to Reproduce**:

1. write code:
```
class Program
{
	static void Main()
	{
		var someBool = new System.Random().Next(0, 2) == 1;
		_ = someBool ? SomeVoid() : default;
	}

	public static void SomeVoid()
	{
		return;
	}
}
```
2. Observe that, at least in Visual Studio, the code looks okay. I think Visual Studio uses Roslyn to analyze, so that's why I'm posting this issue here.
3. Try to build.

**Expected Behavior**:
Visual Studio should show that the code is unacceptable, as a value of type 'void' may not be assigned.

**Actual Behavior**:
Building fails with message "Failed to emit module '...'".

**Comment by nl-x** [comment]
To clarify, the discard (underscore variable) is unrelated. Using `var foo` or even `int foo` instead of the underscore gives the same issue.

The thing is I didn't care about what was going to be returned. My actual code was  
 `_ = connection.Closed ? connection.Open() : default;`.

I tried to mimic the one-liner syntax that I know of  
`_ = someVariable ?? throw new System.ArgumentNullException(nameof(someVariable));`

**Comment by RikkiGibson** [comment]
This appears to be fixed by #40465. Thank you for reporting.
